src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -1646,111 +1644,106 @@ bool CmdSketcherSnap::isActive()
-
-            newItem = new QListWidgetItem;
-            newItem->setData(Qt::UserRole, QVariant(lowid));
-            newItem->setText(lowid == 1       ? tr("Normal Geometry")
-                                 : lowid == 2 ? tr("Construction Geometry")
-                                              : tr("External Geometry"));
-            list->insertItem(2, newItem);
-        }
-    }
+    auto hGrp = getParameterPath();
 
-    void languageChange()
-    {
-        updateWidget();
-    }
+    // 1->Normal Geometry, 2->Construction, 3->External
+    int topid = hGrp->GetInt("TopRenderGeometryId", 1);
+    int midid = hGrp->GetInt("MidRenderGeometryId", 2);
+    int lowid = hGrp->GetInt("LowRenderGeometryId", 3);
 
-protected:
-    QWidget* createWidget(QWidget* parent) override
     {
-        list = new QListWidget();
-        list->setDragDropMode(QAbstractItemView::InternalMove);
-        list->setDefaultDropAction(Qt::MoveAction);
-        list->setSelectionMode(QAbstractItemView::SingleSelection);
-        list->setDragEnabled(true);
-
-        QWidget* renderingWidget = new QWidget(parent);
-        auto* layout = new QGridLayout(renderingWidget);
-        layout->addWidget(list, 0, 0, 0, 0);
-
-        languageChange();
-
-        // Handle change in the order of the list entries
-        QObject::connect(list->model(),
-                         &QAbstractItemModel::rowsMoved,
-                         [this](const QModelIndex& sourceParent,
-                                int sourceStart,
-                                int sourceEnd,
-                                const QModelIndex& destinationParent,
-                                int destinationRow) {
-                             Q_UNUSED(sourceParent)
-                             Q_UNUSED(sourceStart)
-                             Q_UNUSED(sourceEnd)
-                             Q_UNUSED(destinationParent)
-                             Q_UNUSED(destinationRow)
-
-                             int topid = list->item(0)->data(Qt::UserRole).toInt();
-                             int midid = list->item(1)->data(Qt::UserRole).toInt();
-                             int lowid = list->item(2)->data(Qt::UserRole).toInt();
-
-                             auto hGrp = getParameterPath();
-
-                             hGrp->SetInt("TopRenderGeometryId", topid);
-                             hGrp->SetInt("MidRenderGeometryId", midid);
-                             hGrp->SetInt("LowRenderGeometryId", lowid);
-                         });
-
-        return renderingWidget;
+        QSignalBlocker block(this);
+        list->clear();
+
+        QListWidgetItem* newItem = new QListWidgetItem;
+        newItem->setData(Qt::UserRole, QVariant(topid));
+        newItem->setText(topid == 1       ? tr("Normal geometry")
+                                : topid == 2 ? tr("Construction geometry")
+                                            : tr("External geometry"));

```

Comment:
that condition is unreadable and repeated through the code - please extract it to some local lambda or private function.